### PR TITLE
SCR-72: Don't set package metadata to None during an upsert operation

### PR DESCRIPTION
**Pull request by djsauble**
Scraping is a two-phase operation: 1) we scrape Package names from GitHub, then 2) we scrape Package metadata (downloads, categories, etc) from npmjs.com.

Both of these phases used the `insertToPackages` method to update the database. However, since the data from GitHub lacks metadata information, we were passing None values to `insertToPackages`, causing packages to temporarily lose metadata until the phase 2 process repopulated them again (about a 4 hour window).

This PR splits the `insertToPackages` method into two. The `insertToPackages` is now only responsible for inserting package names, and used only in phase 1. The new `updatePackageMetadata` method is solely responsible for setting package metadata, and is only used in phase 2. This fixes the problem described above.

### CODE REVIEW

File: pipeline/scraper/GitHubQuery.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def writeDB(db, result):
                                 dependencyStr = 'pkg:npm/' + k + "@" + result.group()
                             else:
                                 continue
-                        package_id = PSQL.insertToPackages(db, dependencyStr, None, None, None)
+                        package_id = PSQL.insertToPackages(db, dependencyStr)
```
Comment (vpopil) [inline pipeline/scraper/GitHubQuery.py:73]: will probably cause conflict for my branch, haha

File: pipeline/scraper/test.py [hunk 2]
```diff
@@ -85,6 +82,48 @@ def test_insertToPackages(self):
         self.assertTrue(package_name == name)
 
 
+    @ordered
+    def test_updatePackageMetadata(self):
+        db = PSQL.connectToDB()
```
Comment (vpopil) [inline pipeline/scraper/test.py:87]: this potentially (and may some other preps) could be in setUp()

File: pipeline/scraper/test.py [hunk 3]
```diff
@@ -85,6 +82,48 @@ def test_insertToPackages(self):
...
+        name = "myPkg"
+        downloads_last_month = 200
+        categories = ["critical", ",,comma", "\\{braces\\}", "\'quoted\""]
+        modified = datetime.datetime.now()
+
+        # Insert package into the table
+        id = PSQL.insertToPackages(db, name)
+        self.assertTrue(type(id) == int)
```
Comment (vpopil) [inline pipeline/scraper/test.py:95]: `isinstance(id, int)`

File: pipeline/scraper/test.py [hunk 4]
```diff
@@ -85,6 +82,48 @@ def test_insertToPackages(self):
...
+
+        # Update metadata in the table
+        PSQL.updatePackageMetadata(db, name, downloads_last_month, categories, modified)
+
+        # Ensure that the modified field is now not None
+        cur.execute(
+            "SELECT modified FROM packages WHERE id = %s;" % (id)
+        )
+        modified_date = cur.fetchone()[0]
+        self.assertTrue(modified_date != None)
```
Comment (vpopil) [inline pipeline/scraper/test.py:113]: `self.assertIsNotNone`

File: pipeline/scraper/test.py [hunk 5]
```diff
@@ -85,6 +82,48 @@ def test_insertToPackages(self):
...
+        # Upsert the same package into the table again
+        id = PSQL.insertToPackages(db, name)
+        self.assertTrue(type(id) == int)
+
+        # Ensure that the modified field is still not None
+        cur.execute(
+            "SELECT modified FROM packages WHERE id = %s;" % (id)
+        )
+        modified_date = cur.fetchone()[0]
+        self.assertTrue(modified_date != None)
```
Comment (vpopil) [inline pipeline/scraper/test.py:124]: assertIsNotNone

File: pipeline/scraper/PSQL.py [hunk 6]
```diff
@@ -4,7 +4,8 @@
 import re
 
 INSERT_TO_APPLICATION_SQL = "INSERT INTO applications (url, name, followers, retrieved, hash) VALUES (%s, %s, %s, %s, %s) ON CONFLICT ON CONSTRAINT unique_url DO UPDATE SET (url, name, followers, retrieved, hash) = (EXCLUDED.url, EXCLUDED.name, EXCLUDED.followers, EXCLUDED.retrieved, EXCLUDED.hash) RETURNING id;"
-INSERT_TO_PACKAGES_SQL = "INSERT INTO packages (name, downloads_last_month, categories, modified, retrieved) VALUES (%s, %s, %s, %s, %s) ON CONFLICT(name) DO UPDATE SET (name, downloads_last_month, categories, modified) = (EXCLUDED.name, EXCLUDED.downloads_last_month, EXCLUDED.categories, EXCLUDED.modified) RETURNING id;"
+INSERT_TO_PACKAGES_SQL = "INSERT INTO packages (name, retrieved) VALUES (%s, %s) ON CONFLICT(name) DO UPDATE SET (name, retrieved) = (EXCLUDED.name, EXCLUDED.retrieved) RETURNING id;"
+UPDATE_PACKAGE_METADATA_SQL = "UPDATE packages SET downloads_last_month = %s, categories = %s, modified = %s WHERE name = %s;"
```
Comment (vpopil) [inline pipeline/scraper/PSQL.py:8]: would be nice to break into more lines these long ones (as SQL clauses go)

File: pipeline/scraper/test.py (hunk 3 above)
Comment (djsauble) [reply pipeline/scraper/test.py:95]: Fixed

File: pipeline/scraper/test.py (hunk 4 above)
Comment (djsauble) [reply pipeline/scraper/test.py:113]: Fixed

File: pipeline/scraper/test.py (hunk 5 above)
Comment (djsauble) [reply pipeline/scraper/test.py:124]: Fixed

File: pipeline/scraper/PSQL.py (hunk 6 above)
Comment (djsauble) [reply pipeline/scraper/PSQL.py:8]: Fixed

**Pull request merged**